runelite-client/src/main/java/net/runelite/client/plugins/slayer/TargetWeaknessOverlay.java

```diff
@@ -0,0 +1,159 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.slayer;
+
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.util.List;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.NPC;
+import net.runelite.api.Perspective;
+import net.runelite.api.Point;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.game.NPCManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayUtil;
+import net.runelite.client.util.Text;
+
+class TargetWeaknessOverlay extends Overlay
+{
+	private final Client client;
+	private final SlayerConfig config;
+	private final SlayerPlugin plugin;
+	private final ItemManager itemManager;
+	private final NPCManager npcManager;
+
+	@Inject
+	private TargetWeaknessOverlay(Client client, SlayerConfig config, SlayerPlugin plugin, ItemManager itemManager, NPCManager npcManager)
+	{
+		this.client = client;
+		this.config = config;
+		this.plugin = plugin;
+		this.itemManager = itemManager;
+		this.npcManager = npcManager;
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!config.weaknessPrompt())
+		{
+			return null;
+		}
+
+		Task curTask = Task.getTask(plugin.getTaskName());
+		if (curTask == null || curTask.getWeaknessThreshold() < 0 || curTask.getWeaknessItem() < 0)
+		{
+			return null;
+		}
+
+		int threshold = curTask.getWeaknessThreshold();
+		BufferedImage image = itemManager.getImage(curTask.getWeaknessItem());
+
+		if (image == null)
+		{
+			return null;
+		}
+
+		List<NPC> targets = plugin.getHighlightedTargets();
+		for (NPC target : targets)
+		{
+			int currentHealth = calculateHealth(target);
+
+			if (currentHealth > 0 && currentHealth <= threshold)
+			{
+				renderTargetItem(graphics, target, image);
+			}
+		}
+		return null;
+	}
+
+
+	/**
+	 * Based on how health is calculated in OpponentInfoOverlay
+	 */
+	private int calculateHealth(NPC target)
```

Comment:
Is this really needed here? probably not, it does not need to be that accurate

Reply:
It's a way of dealing with superiors and alternatives having different max health and therefore a different health ratio threshold, even though the numeric threshold is the same.
Although alternative suggestions are welcome, I could for example compare ids/level and hardcode the threshold for each one.

Reply:
Then all you need is hp and divide by ratio. you dont need the rest of the logic